Modules/_abc.c

```diff
@@ -814,23 +843,51 @@ _abc__abc_subclasscheck_impl(PyObject *module, PyObject *self,
         if (scls == NULL) {
             goto end;
         }
+
+        _abc_data *scls_impl;
+        int scls_is_abc = _get_optional_impl(state, scls, &scls_impl);
+        if (scls_is_abc < 0) {
+            goto end;
```

Comment:
Sorry I missed the fact that `scls` should be decrefed here as well.

Reply:
Fixed